src/DoggyFrictions.ExternalApi/Controllers/ActionsController.cs

```diff
@@ -48,6 +49,12 @@ public async Task<IActionResult> Post(string sessionId, [FromForm] ActionObject
         {
             return BadRequest();
         }
+        
+        var totalSpent = actionObject.Consumptions.Sum(consumption => consumption.Amount);
```

Comment:
This code is duplicated in this class. Let's push it down to a service. Also, I'd not be returning plain Bad Request and go with ProblemDetails